### DISCUSSION TITLE
read variable in a function block

### DISCUSSION THREAD

**Discussion by jialiang-yin** [gijzelaerr/python-snap7#305] [Q&A] [answered]
Hello Everyone,
I'd like to ask a question about reading variables, that are saved inside a function block(FB). I have tried reading variable from Data block. Is it also possible to read variables from FB? Which function or methode should I use to get them? 

Thank you!

**Comment by jialiang-yin** [comment, marked as answer]
I have figured out, that the variables inside a function block(DB) will be mapped to a data block. 
Then the variables will be saved, for later usage. So it would be same way to read the variable as the global data block.

**Comment by gijzelaerr** [comment]
Closing as resolved -- the original poster found the answer: variables inside function blocks are mapped to instance data blocks and can be read using db_read() the same way as global data blocks.
